src/dbus/methods/player_peer_info.c

```diff
@@ -34,38 +35,90 @@ void cmd_peer_player_info_dbus_cb(const char *result,
 {
     struct cb_args *a = (struct cb_args *) args;
 
+    GVariantBuilder builder;
+    GVariant *playerinfodict;
+
+    g_variant_builder_init (&builder, G_VARIANT_TYPE("a{?*}"));
+
     if (result == NULL)
     {
-      result = "-1";
+        g_variant_builder_add (&builder, "{ss}", "error", "-1");
+    }
+    else
+    {
+        g_variant_builder_add(&builder, "{ss}", "online_id", get_info(result, "online_id='", "'", NULL) ? : "");
```

Comment:
`g_variant_builder_add` takes a `const char *`. It means it will not free up memory returned by `get_info`.

If you really want to try returning a dictionary:
 1. Create a structure in `wb_xmpp_wf.h` e.g. `struct player_peer_info`;
 2. Put all `get_info` parsing inside `xmpp_wf/get/xmpp_iq_peer_player_info.c`;
 3. `xmpp_iq_peer_player_info.c` will give to its callback the filled up structure as const `const struct player_peer_info *` ;
 4. Then `cmd_peer_player_info_dbus_cb` can use the fields of the structure;
 5. Then `xmpp_iq_peer_player_info.c` will free up all fields in the structure;

e.g.
```
s.nickname = get_info(...);
s.banner_badge = get_info_int(...);

if (a->cb)
{
  a->cb(&s);
}

free(s.nickname);
[...]
```

```
        g_variant_builder_add(&builder, "{ss}", "nickname", s->nickname);
        g_variant_builder_add(&builder, "{si}", "banner_badge", s->banner_badge);
        [...]
```

PS: This change will slightly affect `cmd_safe.c`, but it's for the better anyway (deleting few lines).